src/components/RigidBody.vue

```diff
@@ -1,64 +1,82 @@
 <script setup lang="ts">
 import { type TresObject, useLoop } from '@tresjs/core'
-import { shallowRef, watch } from 'vue'
-import { useRapierContext } from '../composables/useRapier'
-import { createCollidersFromChildren } from '../utils/collider'
-import { createRigidBody } from '../utils/rigid-body'
-import type { CreateColliderReturnType } from '../types/collider'
-import type { CreateRigidBodyReturnType, RigidBodyProps } from '../types/rigid-body'
+import { Object3D } from 'three'
+import { onUnmounted, provide, shallowRef, watch } from 'vue'
+
+import { useRapierContext } from '../composables'
+import { createColliderPropsFromObject, createRigidBody } from '../utils'
+import { BaseCollider } from './colliders'
+import type { ColliderProps, RigidBodyContext, RigidBodyProps } from '../types'
 
 const props = withDefaults(defineProps<Partial<RigidBodyProps>>(), {
   type: 'dynamic',
   collider: 'cuboid',
 })
 
-const parentObject = shallowRef<TresObject>()
-const rigidBodyInfos = shallowRef<CreateRigidBodyReturnType>()
-const colliderInfos = shallowRef<CreateColliderReturnType[]>()
-const rigidBodyInstance = shallowRef()
+const { onBeforeRender } = useLoop()
+const { world } = useRapierContext()
 
-defineExpose({
```

Comment:
This is the most important you delete the defineExpose, if you do this, we can apply any methods like force impulse etc, or have access to the instance (the demo of apply force is broken)

This is the current structure, 
 instance: rigidBodyInstance =  This object should be the RigidBody central to apply the methods (couldn't be an array, except in InstanceRigidBody)
 
  rigidBodyInfos, = this should contain all the information about the instance, (object, RigidBody, RigidBodyDesc, Collider)
  collider: colliderInfos,  = This should contain the colliders info (should be an array, right?, since we haev one to many relationship between RigidBody and Colliders)
  group: parentObject, = this is just the template ref of the group
  
  Any question, feel free to reach me.
  
  You can see the implementation in the ApplyForceDemo, to give you an idea what to expose.
  
  Again if the structure make no sense to you, lets discuss

Reply:
Oh my bad, I missed this part when I was resolving conflicts

Let me correct it, yes it make sense to me

Reply:
Please, don't mark the comment as resolved :)

As a good practice (I know is a little inconvenient) only the person doing the review should mark them